src/Middleware/HttpLogging/src/HttpLog.cs

```diff
@@ -6,16 +6,18 @@
 
 namespace Microsoft.AspNetCore.HttpLogging;
 
-internal sealed class HttpRequestLog : IReadOnlyList<KeyValuePair<string, object?>>
+internal sealed class HttpLog : IReadOnlyList<KeyValuePair<string, object?>>
 {
     private readonly List<KeyValuePair<string, object?>> _keyValues;
+    private readonly string _title;
     private string? _cachedToString;
 
-    internal static readonly Func<object, Exception?, string> Callback = (state, exception) => ((HttpRequestLog)state).ToString();
+    internal static readonly Func<object, Exception?, string> Callback = (state, exception) => ((HttpLog)state).ToString();
 
-    public HttpRequestLog(List<KeyValuePair<string, object?>> keyValues)
+    public HttpLog(List<KeyValuePair<string, object?>> keyValues, string title)
     {
         _keyValues = keyValues;
+        _title = title;
```

Comment:
Does this `title` need to be returned in the `KeyValuePair` list too?

Reply:
No, if you're only consuming the KVP's then you can get the title from the event id.